### PR TITLE
UI: Fix collapse/expand all functionality

### LINKED ISSUE #28457: [Bug]: Collapse all broken (`api.collapseAll is not a function`)
### Describe the bug

The "collapse all" action is broken. It logs an error in the console:

```
Uncaught TypeError: api.collapseAll is not a function
```

Screenshot 2024-07-04 at 16 42 42

### Reproduction link

https://635781f3500dd2c49e189caf-hyyihpyiuz.chromatic.com

### Reproduction steps

1. Visit a Storybook
2. Open the main menu (cog icon)
3. Click "collapse all"
4. Check the browser console

### System

```bash
Storybook 8.1.11
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by filipemelo2002**
Closes https://github.com/storybookjs/storybook/issues/28457

Fix: Update Menu "Collapse All" option to emit STORIES_COLLAPSE_ALL 
Issue: Clicking on "Collapse All" throws an error `api.collapseAll is not a function` 

Solution: Updated the `collapse` from the Menus.tsx component so it can emit the new events instead of calling the old methods.

### Testing

1. Visit a Storybook
2. Open the main menu (cog icon)
3. Click "collapse all"
4. Verify the tabs were collapsed

References:

Related issue: [#28457](https://github.com/storybookjs/storybook/issues/28457)
Related PR: [#25486](https://github.com/storybookjs/storybook/pull/25486)

### CODE REVIEW

**Review by shilman — COMMENTED** [review]
There's also a dangling reference to `collapseAll` in `shortcuts.ts`. Can we fix that too?

https://github.com/storybookjs/storybook/blob/9b12305d443cb0136f6a5371957d1375c3bd91c0/code/lib/manager-api/src/modules/shortcuts.ts#L359

File: code/core/src/manager-api/modules/shortcuts.ts [hunk 1]
```diff
@@ -356,7 +356,7 @@ export const init: ModuleFn = ({ store, fullAPI, provider }) => {
           break;
         }
         case 'collapseAll': {
-          fullAPI.collapseAll();
+          fullAPI.emit(STORIES_COLLAPSE_ALL);
```
Comment (ghengeveld) [inline code/core/src/manager-api/modules/shortcuts.ts:359]: There's also `expandAll` which needs a similar update on line 363.

**Review by ghengeveld — APPROVED** [review]
Looks good, just a minor comment to handle `expandAll` as well.

**Review by ghengeveld — APPROVED** [review]
Nice, thank you!

**Pull request merged**